api/v1alpha1/author_webhook.go

```diff
@@ -0,0 +1,167 @@
+	}
+}
+
+// handleRelease takes an incoming admission request and returns an admission response. Create requests
+// add an author label with the current user. Update requests are rejected if the author label is being
+// modified. All other requests are accepted without action.
+func (a *authorWebhook) handleRelease(req admission.Request) admission.Response {
+	release := &Release{}
+	err := json.Unmarshal(req.Object.Raw, release)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, errors.Wrap(err, "error decoding object"))
+	}
+
+	switch req.AdmissionRequest.Operation {
+	case admissionv1.Create:
+		a.setAuthorLabel(req.UserInfo.Username, release)
+
+		return a.patchResponse(req.Object.Raw, release)
+	case admissionv1.Update:
+		oldRelease := &Release{}
+		err := json.Unmarshal(req.OldObject.Raw, oldRelease)
+		if err != nil {
+			return admission.Errored(http.StatusBadRequest, errors.Wrap(err, "error decoding object"))
+		}
+
+		if release.GetLabels()[metadata.AuthorLabel] != oldRelease.GetLabels()[metadata.AuthorLabel] {
+			return admission.Errored(http.StatusBadRequest, errors.New("release author label cannnot be updated"))
+		}
+	}
+	return admission.Allowed("Success")
+}
+
+// handleReleasePlan takes an incoming admission request and returns an admission response. If the
+// attribution label is set to true, the current user is set as the author. If the attribution label
+// is false, the author label is removed. The only exception is if the attribution label remains true
+// during an update and the author value is not modified, the previous author label remains.
+func (a *authorWebhook) handleReleasePlan(req admission.Request) admission.Response {
+	releasePlan := &ReleasePlan{}
+	err := json.Unmarshal(req.Object.Raw, releasePlan)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, errors.Wrap(err, "error decoding object"))
+	}
+	// Author label should not exist in any case if attribution is not true
+	if releasePlan.GetLabels()[metadata.AttributionLabel] != "true" {
+		delete(releasePlan.GetLabels(), metadata.AuthorLabel)
+	}
+
+	switch req.AdmissionRequest.Operation {
+	case admissionv1.Create:
+		if releasePlan.GetLabels()[metadata.AttributionLabel] == "true" {
+			a.setAuthorLabel(req.UserInfo.Username, releasePlan)
+		}
+	case admissionv1.Update:
+		oldReleasePlan := &ReleasePlan{}
+		err := json.Unmarshal(req.OldObject.Raw, oldReleasePlan)
+		if err != nil {
+			return admission.Errored(http.StatusBadRequest, errors.Wrap(err, "error decoding object"))
+		}
+
+		if releasePlan.GetLabels()[metadata.AttributionLabel] == "true" {
+			author := releasePlan.GetLabels()[metadata.AuthorLabel]
+
+			if oldReleasePlan.GetLabels()[metadata.AttributionLabel] != "true" || author == a.sanitizeLabelValue(req.UserInfo.Username) {
+				a.setAuthorLabel(req.UserInfo.Username, releasePlan)
+			} else {
+				// Preserve previous author if update does not involve the author label
```

Comment:
This, at least for me, is so difficult to understand. What about:
```suggestion
				// Preserve previous author if the new author does not match the user making the change
```